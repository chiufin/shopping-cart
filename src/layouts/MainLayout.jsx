import React from 'react'
import PropTypes from 'prop-types'

import styles from './MainLayout.scss'
import Header from '../components/Header/Header'

const MainLayout = ({ children }) => (
  <div data-e2e="app" className="app">
    <a href="#app-content" className={styles['sr-only']}>
      Skip to main content
    </a>
    <Header />
    <div id="app-content" className="app-content">
      <main>
        {children}
      </main>
    </div>
  </div>
)

MainLayout.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object, PropTypes.string]).isRequired,
}

export default MainLayout
