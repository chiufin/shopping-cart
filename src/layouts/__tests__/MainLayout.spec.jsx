import React from 'react'
import { shallow } from 'enzyme'
import MainLayout from '../MainLayout'

jest.mock('./Header.scss', () => ({
  header: 'header',
}))

describe('MainLayout', () => {
  let mountedMainLayout

  beforeEach(() => {
    mountedMainLayout = shallow((
      <MainLayout>
        HELLO
      </MainLayout>
    ))
  })

  it('Renders correctly', () => {
    expect(mountedMainLayout.find('[data-e2e="app"]')).toMatchSnapshot()
  })
})
