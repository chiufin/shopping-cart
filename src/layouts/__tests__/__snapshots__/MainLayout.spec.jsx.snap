// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout Renders correctly 1`] = `
<div
  className="app"
  data-e2e="app"
>
  <a
    href="#app-content"
  >
    Skip to main content
  </a>
  <Header />
  <div
    className="app-content"
    id="app-content"
  >
    <main>
      HELLO
    </main>
  </div>
</div>
`;
